### DISCUSSION TITLE
(ROS Noetic) catkin_make fails, as workspace contains non-ros packages

### DISCUSSION THREAD

**Discussion by RomanStadlhuber** [OpenVSLAM-Community/openvslam_ros#171] [Q&A] [answered]
Hello, I am trying to build `stella_vslam_ros` (from `ros`) branch as detailed in the [ROS Installation Instructions](https://stella-cv.readthedocs.io/en/latest/ros_package.html#installation). Everything works out as expected, except for the last step, that is I get an error at

```bash
mkdir -p ~/catkin_ws/src
cd ~/catkin_ws/src
git clone --recursive -b ros --depth 1 https://github.com/stella-cv/stella_vslam_ros.git
cd ~/catkin_ws/
rosdep install -y -i --from-paths ~/catkin_ws/src --skip-keys=stella_vslam
catkin_make -j
```

specifically, `catkin_make` fails with

```
CMake Error at /opt/ros/noetic/share/catkin/cmake/catkin_workspace.cmake:100 (message):
  This workspace contains non-catkin packages in it, and catkin cannot build
  a non-homogeneous workspace without isolation.  Try the
  'catkin_make_isolated' command instead.
Call Stack (most recent call first):
  CMakeLists.txt:69 (catkin_workspace)
```
I have tried running `catkin_make_isolated` and it works without throwing an error, however when I try to

```bash
rosrun stella_vslam_ros run_slam ...
```

I get an error that the node/executable could not be found

```
[rosrun] Couldn't find executable named run_slam below /home/user/catkin_ws/src/stella_vslam_ros
```

What could be going wrong here?

I was unable to find a similar issue/question/discussion to this, so I am sorry if this is a duplicate.
If necessary, I will happily provide additional information.

Kind regards.

**Comment by ymd-stella** [comment]
Is there any WARNING?
Please show me the complete log of catkin_make.

**Reply by RomanStadlhuber** [reply, marked as answer]
Hello, please excuse my late reply. I found that I have accidentally cloned `stella_vslam` into `~/catkin_ws/src` instead of `~/lib`, which was the cause of this issue.
The Problem is thereby resolved.
Thank you very much for your answer.
